BitwardenShared/UI/Vault/Vault/VaultList/VaultListProcessorTests.swift

```diff
@@ -129,6 +135,71 @@ class VaultListProcessorTests: BitwardenTestCase { // swiftlint:disable:this typ
         XCTAssertNil(stateService.loginRequest)
     }
 
+    /// `perform(_:)` with `appeared` does not register the device for notifications
+    /// if the user has denied notifications
+    func test_perform_appeared_notificationRegistration_denied() async throws {
+        stateService.activeAccount = .fixture()
+        notificationService.authorizationStatus = .denied
+
+        await subject.perform(.appeared)
+
+        XCTAssertFalse(application.registerForRemoteNotificationsCalled)
+    }
+
+    /// `perform(_:)` with `appeared` does not register the device for notifications
+    /// if there is an error
+    func test_perform_appeared_notificationRegistration_errored() async throws {
+        stateService.activeAccount = .fixture()
+        notificationService.authorizationStatus = .authorized
+        stateService.notificationsLastRegistrationError = BitwardenTestError.example
+
+        await subject.perform(.appeared)
+
+        XCTAssertFalse(application.registerForRemoteNotificationsCalled)
+        XCTAssertEqual(errorReporter.errors as? [BitwardenTestError], [.example])
+    }
+
+    /// `perform(_:)` with `appeared` does not register the device for notifications
+    /// if the device attempted registration less than one day (that is, 86400 seconds) ago.
+    func test_perform_appeared_notificationRegistration_lessThanADay() async throws {
+        stateService.activeAccount = .fixture()
+        notificationService.authorizationStatus = .authorized
+        stateService.notificationsLastRegistrationDates["1"] = timeProvider.presentTime.addingTimeInterval(-86399)
+
+        await subject.perform(.appeared)
+
+        XCTAssertFalse(application.registerForRemoteNotificationsCalled)
+        XCTAssertEqual(
+            stateService.notificationsLastRegistrationDates["1"],
+            timeProvider.presentTime.addingTimeInterval(-86399)
+        )
+    }
+
+    /// `perform(_:)` with `appeared` registers the device for notifications
+    /// if the device attempted registration more than one day (that is, 86400 seconds) ago.
+    func test_perform_appeared_notificationRegistration_moreThanADay() async throws {
```

Comment:
🎨 If I'm not mistaken this would be to exactly a day. Can another test be added for more than a day? i.e. 86401 so we have the three possible cases for `>=`.

Reply:
I was mostly trying to test the boundary conditions, but I can add another one explicitly for 86401

Reply:
Great thanks! Just in case my reasoning behind this is because on a given `x >= n` condition, if we don't add a test for `x > n` then all the tests will keep passing if the condition would change to `x == n`.

Reply:
It makes sense, though the tests would still pass if the condition changed to `x == n || x == n+1`. Figuring out the optimal set of tests to cover as much as possible without creating so many tests that execution takes forever is a tricky thing.